test/utils/urlEncoding/decodeSymbol.spec.ts

```diff
@@ -0,0 +1,24 @@
+import { decodeSymbol } from '../../../src/utils'
+describe('Decoding symbol', () => {
+  test('A-Z symbol', () => {
+    expect(decodeSymbol('TNK')).toBe('TNK')
+  })
+
+  test('Special chars', () => {
+    expect(decodeSymbol("Token%20%2F21%20%_%2055%3F%20**%20%7C%_%3D%2B%23%40%25%5E%26%60'%22")).toBe(
+      'Token /21 - 55? ** |-=+#@%^&`\'"',
+    )
+  })
+  test('Dash in the symbol', () => {
+    expect(decodeSymbol('TN%_HUI')).toBe('TN-HUI')
+  })
+  test('Leading and trailing spaces', () => {
+    expect(decodeSymbol('%20SPACES%20HERE%20%20')).toBe('SPACES HERE')
+  })
+  test('Unicode chars', () => {
+    expect(decodeSymbol('Oh%20%F0%9F%92%BB%20%F0%9F%A7%B5%20%F0%9F%94%A5')).toBe('Oh 💻 🧵 🔥')
```

Comment:
This is more realistic, actually, I'm going to mint a token myself with emojis

Reply:
Just don't call it Emo Token :joy:

Reply:
I dare you!